src/Symfony/Bundle/FrameworkBundle/Controller/RedirectController.php

```diff
@@ -30,11 +31,11 @@ class RedirectController
     private $httpPort;
     private $httpsPort;
 
-    public function __construct(UrlGeneratorInterface $router = null, int $httpPort = null, int $httpsPort = null)
+    public function __construct(RequestContext $requestContext, UrlGeneratorInterface $router = null)
```

Comment:
This change to cover the config in [router.xml:120](https://github.com/symfony/symfony/pull/37280/files#diff-a54ffdcb2ee5f6c0c06492fabab70827L120). Many thanks if you have a better way to do it.

Reply:
you can achieve the same in the php config. you need to use `inline_service`

Reply:
Thanks @Tobion. Updated.